### PR TITLE
Add Github PR Template

### PR DESCRIPTION
**Pull request by matthewrsj**
We have been using a pull request template in IAM for a while now (thanks @ElijahCaine for introducing that). I was skeptical at first but it is extremely helpful, especially for including the steps necessary to test the PR. As a reviewer who isn't actively working on the codebase, I would love to see this.

@leian7 @athai @Kennric

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  This is a guideline for what a PR should look like.
+  Feel free to modify it to fit your specific needs.
+
+  Please list the issue this fixes in the PR
+-->
+fixes issue #xx
+
+## Changes in this PR.
+<!--
+  Please include a lsit of all things this PR will accomplish
```
Comment (athai) [inline .github/PULL_REQUEST_TEMPLATE.md]: 'list' is spelled incorrectly

**Comment by athai** [comment]
+1 I'm all for this; it seems like a great idea.

**Comment by ElijahCaine** [comment]
We should consider adding an 'expected output' section to the template like we are adding to IaM.

**Comment by leian7** [comment]
+1

**Comment by ElijahCaine** [comment]
+1.
I wonder if we should add a note to the expected output to only post parts of the output that pertain to the PR (specific tests passing/failing, etc)....

**Comment by matthewrsj** [comment]
@ElijahCaine hopefully that is fairly obvious. It should be assumed that the developer making the PR is keeping things in the PR in the scope of the PR.... PR PR PR

**Pull request merged**